tracing-journald/src/socket.rs

```diff
@@ -31,18 +33,38 @@ fn cmsg_buffer_size_for_one_fd() {
     assert_cmsg_bufsize()
 }
 
-pub fn send_one_fd(socket: &UnixDatagram, fd: RawFd) -> Result<usize> {
+pub fn send_one_fd_to<P: AsRef<Path>>(socket: &UnixDatagram, fd: RawFd, path: P) -> Result<usize> {
     assert_cmsg_bufsize();
 
-    let mut cmsg_buffer = AlignedBuffer {
-        buffer: ([0u8; CMSG_BUFSIZE]),
+    let mut addr: sockaddr_un = unsafe { zeroed() };
+    let path_bytes = path.as_ref().as_os_str().as_bytes();
+    // path_bytes may have at most sun_path + 1 bytes, to account for the trailing NUL byte.
+    if addr.sun_path.len() <= path_bytes.len() {
+        return Err(Error::from_raw_os_error(ENAMETOOLONG));
+    }
+
+    addr.sun_family = AF_UNIX as _;
+    unsafe {
```

Comment:
Not sure how to do this in safe Rust, given that I've got a `u8` slice in Rust land and a `c_char` = `i8` array in C land… :thinking:

Reply:
You could write the for loop explicitly (LLVM is supposed to be pretty good at converting such loops into `memcpy` calls anyway) but I think this is fine, in context.